SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/CodeGenerator/Base/TypeDescription.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Improbable.CodeGen.Base
+{
+    [DebuggerDisplay("{QualifiedName.QualifiedName}")]
+    public readonly struct TypeDescription
+    {
+        private readonly Bundle bundle;
+
+        public readonly string Namespace;
+
+        public readonly string Name;
+
+        public readonly string QualifiedName;
+
+        public readonly IReadOnlyList<TypeDescription> NestedTypes;
+
+        public readonly IReadOnlyList<EnumDefinition> NestedEnums;
+
+        public readonly IReadOnlyList<FieldDefinition> Fields;
+
+        public readonly IReadOnlyList<Annotation> Annotations;
+
+        public readonly IReadOnlyList<ComponentDefinition.EventDefinition> Events;
+
+        public readonly bool IsNestedType;
+
+        public readonly bool IsComponent;
+        public readonly uint? ComponentId;
+        
+        public TypeDescription(string qualifiedName, Bundle bundle)
+        {
+            this.bundle = bundle;
+
+            Name = Text.SnakeCaseToPascalCase(qualifiedName).Split('.').Last();
+            Namespace = $"{Text.GetNamespaceFromTypeName(qualifiedName)}";
+            QualifiedName = qualifiedName;
+
+            var directNestedTypes = bundle.GetNestedTypes(qualifiedName);
+            var directNestedEnums = bundle.GetNestedEnums(qualifiedName);
+
+            NestedTypes = directNestedTypes.Select(id =>
+            {
+                var t = bundle.Types[id];
+                return new TypeDescription(t.Identifier.QualifiedName, bundle);
+            }).ToList();
+
+            NestedEnums = directNestedEnums.Select(id => bundle.Enums[id]).ToList();
+
+            IsNestedType = bundle.IsNestedType(qualifiedName);
+
+            IsComponent = bundle.Components.TryGetValue(qualifiedName, out var component);
+            ComponentId = component?.ComponentId;
```

Comment:
Can we check against the accepted range (1000-2000?) here or somewhere else so that it doesn't trip up the user?